clock/testing/fake_clock.go

```diff
@@ -231,6 +249,12 @@ func (*IntervalClock) Tick(d time.Duration) <-chan time.Time {
 	panic("IntervalClock doesn't implement Tick")
 }
 
+// NewTicker is currently unimplemented, will panic.
+// TODO: make interval clock use FakeClock so this can be implemented.
+func (*IntervalClock) NewTicker(d time.Duration) clock.Ticker {
```

Comment:
Why do we need this function again? If we didn't, IntervalClock wouldn't implement the "WithTicker" interface, which means that it would fail at compile time. By adding it, it now implements WithTicker, but will fail at runtime?